src/abstractdataframe/selection.jl

```diff
@@ -592,8 +592,9 @@ function _gen_colnames(@nospecialize(res), newname::Union{AbstractVector{Symbol}
 
     if newname !== AsTable && newname !== nothing
         if length(colnames) != length(newname)
-            throw(ArgumentError("Number of returned columns does not match the " *
-                                "length of requested output"))
+            throw(ArgumentError("Number of returned columns is $(length(colnames)) " *
```

Comment:
We could print `res` as a whole but it is probably not very useful.

Reply:
Of course what you as is potentially doable, but it would be quite complex, so I am hesitant and hope that passing number of columns is an enough improvement.

Reply:
That makes sense to me.